core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -333,6 +333,9 @@ object TestUtils extends Logging {
     if (!props.containsKey(GroupCoordinatorConfig.GROUP_INITIAL_REBALANCE_DELAY_MS_CONFIG))
       props.put(GroupCoordinatorConfig.GROUP_INITIAL_REBALANCE_DELAY_MS_CONFIG, "0")
     rack.foreach(props.put(ServerConfigs.BROKER_RACK_CONFIG, _))
+    // reduce log cleaner offset map memory usage, must be at greater than 1MB per cleaner thread, set to 2M+2 so that
+    // we can set 2 cleaner threads.
+    props.put(CleanerConfig.LOG_CLEANER_DEDUPE_BUFFER_SIZE_PROP, "2097154")
```

Comment:
Is this change related to the PR?

Reply:
Yes, I update this config from 1 to 2 in my test cases, which would fail without this change.

Reply:
Is this something that we could set at the test level? You can pass extra configs to `@ClusterTest`.

Reply:
This approach is more reasonable, I reverted the changes in TestUtils and KafkaClusterTestKit, but we need to change `props.put` to `props.putIfAbsent` in  KafkaClusterTestKit since it will overwrite the original value.
Outdated